packages/core/src/StateNode.ts

```diff
@@ -1888,6 +1889,18 @@ class StateNode<
       internal,
       eventType: transitionConfig.event
     };
+
+    Object.defineProperty(transition, 'toJSON', {
+      value: () => ({
+        ...transition,
+        target: transition.target
+          ? transition.target.map(t => `#${t.id}`)
+          : undefined,
+        source: `#{this.id}`
```

Comment:
isn't this redundant? It shouldn't be needed for rehydration as this always points to the node on which transitions are defined and this cannot be changed or manipulated anyhow.

Reply:
It's necessary when aggregating transitions, similar to how `event` is also defined.

Reply:
Could you illustrate this with an example?